### PR TITLE
refactor: tables query action

### PR DESCRIPTION
**Pull request by fontanierh**
Bring back the structured DB query action, but with the new data model.

Also:
- Delete the remaining database-related code (will then delete the actual tables and fields)
- Assistant builder now allows to select multiple tables

Not done:
- Assistant builder (root level) doesn't display the table names (and datasource names)
- Upload CSV not yet moved in the folder page
I'll address these in a subsequent PR (this one is already a bit involved)

Deploy plan: run init DB on front + deploy

### CODE REVIEW

**Comment by spolu** [comment]
(please update deploy plan :))

Will review maybe tonight surely tomorrow morning :+1:

**Review by spolu — CHANGES_REQUESTED** [review]
LGTM

Left a few minor comments + the ask to split the dataSourceConfig construction in another PR.

File: front/lib/api/assistant/configuration.ts [hunk 1]
```diff
@@ -277,64 +273,67 @@ export async function getAgentConfigurations<V extends "light" | "full">({
       : Promise.resolve({} as Record<string, AgentUserRelation>),
   ]);
 
-  const agentDatasourceConfigurations = (
+  const agentDatasourceConfigurationsPromise = (
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:276]: Yep, will split out

File: front/lib/api/assistant/configuration.ts [hunk 2]
```diff
@@ -277,64 +273,67 @@ export async function getAgentConfigurations<V extends "light" | "full">({
...
+  ).then((dsConfigs) =>
+    dsConfigs.reduce((acc, dsConfig) => {
+      acc[dsConfig.retrievalConfigurationId] =
+        acc[dsConfig.retrievalConfigurationId] || [];
+      acc[dsConfig.retrievalConfigurationId].push(dsConfig);
+      return acc;
+    }, {} as Record<number, AgentDataSourceConfiguration[]>)
+  );
+
+  const agentTablesConfigurationTablesPromise = (
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:307]: yes it will only run for full (`Object.values(tablesQueryConfigs).length` is equal to 0 otherwise)

File: front/lib/models/index.ts [hunk 3]
```diff
@@ -47,8 +43,6 @@ import { XP1Run, XP1User } from "@app/lib/models/xp1";
 
 export {
   AgentConfiguration,
-  AgentDatabaseQueryAction,
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/models/index.ts:50]: Missed that part (although I'm not convinced we are getting much out of this index file, the goal of this PR isn't to change this)

File: front/pages/w/[wId]/builder/assistants/[aId]/index.tsx [hunk 4]
```diff
@@ -108,13 +109,30 @@ export async function buildInitialState({
     }
   }
 
-  const databaseQueryConfiguration: AssistantBuilderInitialState["databaseQueryConfiguration"] =
-    null;
+  let tablesQueryConfiguration: AssistantBuilderInitialState["tablesQueryConfiguration"] =
+    {};
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/pages/w/[wId]/builder/assistants/[aId]/index.tsx:113]: > Why depart from null? Would love to see the approach aligned between actions?

It is aligned with the `dataSources` action. These 2 are similar, we need an object because you can add/remove specific sources (and you can have several). 
Makes the builder code simpler to just initialize right away with the correct thing instead of allowing null and having to check it everywhere.

File: types/src/front/assistant/actions/tables_query.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    dataSourceId: string;
+    tableId: string;
+  }>;
+};
+
+export type TablesQueryActionType = {
+  id: ModelId;
+  type: "tables_query_action";
+
+  params: Record<string, string | number | boolean>;
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply types/src/front/assistant/actions/tables_query.ts:18]: > We have a DustAppParameters

Will use that 👍  thanks

> But can't they be more typed here?

Not sure we'd want that. We cant guarantee it so any other type will be optimistic / a potential lie / a false sense of safety IMO

File: types/src/front/assistant/actions/tables_query.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  output: Record<string, string | number | boolean> | null;
```
[the remark this answers is not in the document]
Comment (fontanierh) [reply types/src/front/assistant/actions/tables_query.ts:19]: We control the app but we can mess it up too, so I'd rather check ?

File: front/pages/w/[wId]/builder/assistants/[aId]/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (spolu) [reply front/pages/w/[wId]/builder/assistants/[aId]/index.tsx:117]: Gotcha :+1:

File: types/src/front/assistant/actions/tables_query.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (spolu) [reply types/src/front/assistant/actions/tables_query.ts]: Fair. Maybe as we converge. But then the type you propose is a lie as you can have nested structures. Hence the unknown in DustAppRun?

File: front/lib/api/assistant/configuration.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/api/assistant/configuration.ts:310]: ok I see the condition up there. Hard to follow but trusting you.

File: types/src/front/assistant/actions/tables_query.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (fontanierh) [reply types/src/front/assistant/actions/tables_query.ts]: Indeed it's also a lie. I cargo culted from what we had before. I'll give it more thoughts

File: front/lib/api/assistant/configuration.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:310]: The rationale is: 
- fetch `tablesQueryConfigs` if we are in full variant.
- fetch `tablesQueryConfigTables` if we have some `tablesQueryConfigs` (which happens to only be the case if we're in full variant, but not relevant at this stage: if we have some `tablesQueryConfigs` we need to have their tables as well)

**Review by spolu — APPROVED** [review]
LGTM 👍
